hermes-benchmark/src/jmh/java/pl/allegro/tech/hermes/benchmark/environment/FrontendEnvironment.java

```diff
@@ -0,0 +1,102 @@
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.TearDown;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import pl.allegro.tech.hermes.api.Group;
+import pl.allegro.tech.hermes.domain.group.GroupRepository;
+import pl.allegro.tech.hermes.domain.topic.TopicRepository;
+import pl.allegro.tech.hermes.frontend.HermesFrontend;
+import pl.allegro.tech.hermes.test.helper.environment.KafkaStarter;
+import pl.allegro.tech.hermes.test.helper.environment.ZookeeperStarter;
+
+import java.io.IOException;
+
+import static pl.allegro.tech.hermes.api.ContentType.AVRO;
+import static pl.allegro.tech.hermes.test.helper.builder.TopicBuilder.topic;
+
+@State(Scope.Benchmark)
+public class FrontendEnvironment {
+
+    private static final Logger logger = LoggerFactory.getLogger(FrontendEnvironment.class);
+    private static final int MAX_CONNECTIONS_PER_ROUTE = 200;
+    private static final int ZOOKEEPER_PORT = 2181;
+
+    private HermesFrontend hermesFrontend;
+    private ZookeeperStarter zookeeperStarter;
+    private KafkaStarter kafkaStarter;
+    private HermesPublisher publisher;
+    private MetricRegistry metricRegistry;
+
+    public static void main(String[] args) throws Exception {
+        new FrontendEnvironment().setupServers();
+    }
+
+    @Setup(Level.Trial)
+    public void setupServers() throws Exception {
+        zookeeperStarter = new ZookeeperStarter(ZOOKEEPER_PORT, "localhost");
+        zookeeperStarter.start();
+
+        kafkaStarter = new KafkaStarter("/kafka.properties");
+        kafkaStarter.start();
+
+        hermesFrontend = HermesFrontend.frontend().build();
+        hermesFrontend.start();
+
+        GroupRepository groupRepository = hermesFrontend.getService(GroupRepository.class);
+        TopicRepository topicRepository = hermesFrontend.getService(TopicRepository.class);
+        groupRepository.createGroup(Group.from("bench"));
+        topicRepository.createTopic(topic("bench.topic")
+                .withContentType(AVRO)
+                .withMessageSchema(loadMessageResource("schema"))
+                .build());
+    }
+
+    @Setup(Level.Trial)
+    public void setupPublisher() throws Exception {
+        metricRegistry = new MetricRegistry();
+
+        String messageBody = loadMessageResource("completeMessage");
+        publisher = new HermesPublisher(MAX_CONNECTIONS_PER_ROUTE, "http://localhost:8080/topics/bench.topic", messageBody, metricRegistry);
+    }
+
+    @TearDown(Level.Trial)
+    public void shutdownServers() throws Exception {
+        hermesFrontend.stop();
+        kafkaStarter.stop();
+        zookeeperStarter.stop();
+    }
+
+    @TearDown(Level.Trial)
+    public void shutdownPublisherAndReportMetrics() throws Exception {
+        reportMetrics();
+
+        shutdownPublisher();
+    }
+
+    public void shutdownKafka() throws Exception {
```

Comment:
removed, this is because I have other bench tests shelved which use this method, however they are not worthy of publish